src/scenic/core/distributions.py

```diff
@@ -19,1186 +19,1189 @@
 ## Misc
 
 def dependencies(thing):
-	"""Dependencies which must be sampled before this value."""
-	return getattr(thing, '_dependencies', ())
+    """Dependencies which must be sampled before this value."""
+    return getattr(thing, '_dependencies', ())
 
 def needsSampling(thing):
-	"""Whether this value requires sampling."""
-	return isinstance(thing, Distribution) or dependencies(thing)
+    """Whether this value requires sampling."""
+    if isinstance(thing, list):
+        return any(needsSampling(elem) for elem in thing)
```

Comment:
I don't think this is the right way to handle lists: this would claim a list containing a distribution needs sampling, but if you call `sample` on it you'll get an exception. Also of course it works only for lists and not tuples, dicts, etc. This is supposed to be the job of the `toDistribution` function, which we should call whenever receiving a value from the user that could be random. (Right now `toDistribution` doesn't handle all standard container types either -- I think we might have a TODO item about that.)